testutil/aws.go

```diff
@@ -36,12 +47,29 @@ func TaskMetaHandlerFn(t *testing.T, respFn func() string) http.Handler {
 // Because of the environment variable, this is unsafe for running tests in parallel.
 func TaskMetaServer(t *testing.T, handler http.Handler) {
 	ecsMetadataServer := httptest.NewServer(handler)
+
+	// Help detect invalid concurrent servers since this relies on an environment variable.
+	require.Empty(t, os.Getenv(awsutil.ECSMetadataURIEnvVar),
+		"%s already set. TaskMetaServer cannot be used concurrently.", awsutil.ECSMetadataURIEnvVar,
+	)
```

Comment:
This didn't catch any concurrent servers being started. I thought we could keep it anyway as a safeguard / warning?

Reply:
Yes, I think this is a really good check to leave in. It will catch any future attempts to use this concurrently.